src/simulation/test/LoadGeneratorTests.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/SHA.h"
+#include "crypto/SecretKey.h"
+#include "lib/catch.hpp"
+#include "main/Config.h"
+#include "scp/QuorumSetUtils.h"
+#include "simulation/LoadGenerator.h"
+#include "simulation/Topologies.h"
+#include "test/test.h"
+#include "util/Math.h"
+#include <fmt/format.h>
+
+using namespace stellar;
+
+TEST_CASE("Multi-op pretend transactions are valid", "[loadgen]")
+{
+    auto cfg = getTestConfig();
+
+    // 50% of transactions contain 2 ops,
+    // and 50% of transactions contain 3 ops.
+    cfg.LOADGEN_OP_COUNT_FOR_TESTING = {2, 3};
+    cfg.LOADGEN_OP_COUNT_DISTRIBUTION_FOR_TESTING = {1, 1};
+
+    Hash networkID = sha256(cfg.NETWORK_PASSPHRASE);
+    Simulation::pointer simulation =
+        Topologies::pair(Simulation::OVER_LOOPBACK, networkID);
+
+    simulation->startAllNodes();
+    simulation->crankUntil(
+        [&]() { return simulation->haveAllExternalized(3, 1); },
+        2 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+    auto nodes = simulation->getNodes();
+    auto& app = *nodes[0]; // pick a node to generate load
+
+    auto& loadGen = app.getLoadGenerator();
+    loadGen.generateLoad(LoadGenerator::CREATE, 3, 0, 0, 10, 100,
+                         std::chrono::seconds(0), 0);
+    try
+    {
+        simulation->crankUntil(
+            [&]() {
+                // we need to wait 2 rounds in case the tx don't propagate
+                // to the second node in time and the second node gets the
+                // nomination
+                return simulation->haveAllExternalized(5, 2) &&
+                       loadGen.checkAccountSynced(app, true).empty();
+            },
+            3 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
```

Comment:
you should also check that you have the right values for `loadgen.account.created`, `loadgen.payment.native` and `loadgen.pretend.submitted` (otherwise we don't even know if loadgen did anything)

Reply:
Makes sense! I add them.